src/third/redux-persist/getStoredState.js

```diff
@@ -1,98 +1,93 @@
+      onComplete(err);
+      return;
     }
 
-    let persistKeys = allKeys.filter((key) => key.indexOf(keyPrefix) === 0).map((key) => key.slice(keyPrefix.length))
-    let keysToRestore = persistKeys.filter(passWhitelistBlacklist)
+    const persistKeys = allKeys
+      .filter(key => key.indexOf(keyPrefix) === 0)
+      .map(key => key.slice(keyPrefix.length));
+    const keysToRestore = persistKeys.filter(key => whitelist.indexOf(key) !== -1);
 
-    let restoreCount = keysToRestore.length
-    if (restoreCount === 0) complete(null, restoredState)
-    keysToRestore.forEach((key) => {
+    const restoreCount = keysToRestore.length;
+    if (restoreCount === 0) {
+      onComplete(null, restoredState);
+    }
+    keysToRestore.forEach(key => {
       (async () => {
         let err = null;
         let serialized = null;
         try {
-          serialized = await storage.getItem(createStorageKey(key))
+          serialized = await storage.getItem(createStorageKey(key));
         } catch (e) {
-          err = e
+          err = e;
+        }
+        if (err) {
+          logging.warn(err, {
+            message: 'redux-persist/getStoredState: Error restoring data for a key.',
+            key,
+          });
+        } else {
+          restoredState[key] = rehydrate(key, serialized);
+        }
+        completionCount += 1;
+        if (completionCount === restoreCount) {
+          onComplete(null, restoredState);
         }
-        if (err) logging.warn(err, { message: 'redux-persist/getStoredState: Error restoring data for a key.', key})
-        else restoredState[key] = rehydrate(key, serialized)
-        completionCount += 1
-        if (completionCount === restoreCount) complete(null, restoredState)
-      })()
-    })
+      })();
+    });
   })();
 
-  function rehydrate (key, serialized) {
-    let state = null
+  function rehydrate(key, serialized) {
+    let state = null;
 
     try {
-      let data = deserializer(serialized)
-      state = data
+      const data = deserializer(serialized);
+      state = data;
     } catch (err) {
-      logging.warn(err, { message: 'redux-persist/getStoredState: Error restoring data for a key.', key })
+      logging.warn(err, {
+        message: 'redux-persist/getStoredState: Error restoring data for a key.',
+        key,
+      });
     }
 
-    return state
-  }
-
-  function complete (err, restoredState) {
-    onComplete(err, restoredState)
-  }
-
-  function passWhitelistBlacklist (key) {
-    if (whitelist && whitelist.indexOf(key) === -1) return false
-    if (blacklist.indexOf(key) !== -1) return false
-    return true
-  }
-
-  function createStorageKey (key) {
-    return `${keyPrefix}${key}`
+    return state;
   }
 
-  if (typeof onComplete !== 'function' && !!Promise) {
-    return new Promise((resolve, reject) => {
-      onComplete = (err, restoredState) => {
-        if (err) reject(err)
-        else resolve(restoredState)
-      }
-    })
```

Comment:
Hmm, this is actually something where I'd be inclined to take it in the opposite direction: this `getStoredState` shouldn't take a callback at all, and should always just return a Promise. That's because it's a function whose whole job is to eventually call that callback, and now that JS has promises and a nice async/await syntax, it's cleaner to write that in the form of `await getStoredState(config)`.

It looks like the story is that we don't call this `geStoredState` function at all in our code outside this library; but where it is called in `persistStore`, that uses the callback form. So probably it's best to defer refactoring that until we have types on this code.